### DISCUSSION TITLE
Potential support for webdav-push?

### DISCUSSION THREAD

**Discussion by CiaranG** [Kozea/Radicale#1630] [Ideas]
I'm wondering if anyone has yet looked into whether it will be possible to support this proposed webdav-push protocol extension: https://github.com/bitfireAT/webdav-push/

**Comment by pbiering** [comment]
this would require persistent connections from client to server with I assume huge extension on server side (and also clients must be extended).

**Reply by lukeuser** [reply]
> this would require persistent connections from client to server with I assume huge extension on server side (and also clients must be extended).

Surely that's not true, unless I'm mistaken. Radicale wouldn't be acting as the push server, it would simply be sending a message to the push server each time the data is updated. For instance, using ntfy, radicale would simply be POSTing to ntfy.sh (or a self-host).

Cf. https://f-droid.org/2022/12/18/unifiedpush.html

**Reply by pbiering** [reply]
_radicale_ supports already a hook, feel free to use it and create a surrounding prototype of what you require.

**Reply by lukeuser** [reply]
While a hook could be used to achieve the desired functionality, the webdav-push protocol itself is more complex (the server should advertise its support, manage push subscriptions). I don't have the competency to implement this myself. I feel like providing support for this protocol at some point would make a lot of sense, since push support is a notable enhancement—it's up to you and your fellow contributors of course. Have a look at the draft specification perhaps.

https://bitfireat.github.io/webdav-push/draft-bitfire-webdav-push-00.html

**Comment by CiaranG** [comment]
I guess one next step would be to get an idea of the scale of server-side work necessary by looking at the working NextCloud implementation of this [here](https://github.com/bitfireAT/nc_ext_dav_push).

**Comment by symphorien** [comment]
davx5 now supports webdav push https://github.com/bitfireAT/davx5-ose/releases/tag/v4.5-ose so client support is here.

**Comment by SapphoSys** [comment]
Hey all! I recently implemented WebDAV Push in my client, [Chiri](https://github.com/chiriapp/chiri):
> *Chiri is a cross-platform CalDAV-compatible task management app for desktop.*

Screenshot:

<img width="522" height="209" alt="image" src="https://github.com/user-attachments/assets/9462ddf2-8d91-4935-9cff-4cabfce1dbed" />

Unfortunately, WebDAV Push as a feature is not supported in many CalDAV servers :(
To my knowledge, only [RustiCal](https://github.com/lennart-k/rustical/) and [Nextcloud](https://github.com/bitfireAT/nc_ext_dav_push) (with an extension) support it at the moment.

I'd be very interested in seeing this feature in the Radicale CalDAV server! ^^; (and I imagine lots of other people are as well)
